tests/infer/test_compute_downstream_costs.py

```diff
@@ -1,611 +0,0 @@
-        if include_inner_1:
-            with pyro.plate("plate_inner_1", 3) as ind_inner:
-                pyro.sample(
-                    "c1", dist.Bernoulli(p1).expand_by([len(ind_inner), len(ind_outer)])
-                )
-                if flip_c23 and not include_obs:
-                    pyro.sample(
-                        "c3",
-                        dist.Bernoulli(p0).expand_by([len(ind_inner), len(ind_outer)]),
-                    )
-                    pyro.sample(
-                        "c2",
-                        dist.Bernoulli(p1).expand_by([len(ind_inner), len(ind_outer)]),
-                    )
-                else:
-                    pyro.sample(
-                        "c2",
-                        dist.Bernoulli(p0).expand_by([len(ind_inner), len(ind_outer)]),
-                    )
-                    pyro.sample(
-                        "c3",
-                        dist.Bernoulli(p2).expand_by([len(ind_inner), len(ind_outer)]),
-                    )
-        with pyro.plate("plate_inner_2", 4) as ind_inner:
-            pyro.sample(
-                "d1", dist.Bernoulli(p0).expand_by([len(ind_inner), len(ind_outer)])
-            )
-            d2 = pyro.sample(
-                "d2", dist.Bernoulli(p2).expand_by([len(ind_inner), len(ind_outer)])
-            )
-            assert d2.shape == (4, 2)
-            if include_obs:
-                pyro.sample(
-                    "obs",
-                    dist.Bernoulli(p0).expand_by([len(ind_inner), len(ind_outer)]),
-                    obs=torch.ones(d2.size()),
-                )
-
-
-@pytest.mark.parametrize("include_inner_1", [True, False])
-@pytest.mark.parametrize("include_single", [True, False])
-@pytest.mark.parametrize("flip_c23", [True, False])
-@pytest.mark.parametrize("include_triple", [True, False])
-@pytest.mark.parametrize("include_z1", [True, False])
-def test_compute_downstream_costs_big_model_guide_pair(
-    include_inner_1, include_single, flip_c23, include_triple, include_z1
-):
-    guide_trace = poutine.trace(big_model_guide, graph_type="dense").get_trace(
-        include_obs=False,
-        include_inner_1=include_inner_1,
-        include_single=include_single,
-        flip_c23=flip_c23,
-        include_triple=include_triple,
-        include_z1=include_z1,
-    )
-    model_trace = poutine.trace(
-        poutine.replay(big_model_guide, trace=guide_trace), graph_type="dense"
-    ).get_trace(
-        include_obs=True,
-        include_inner_1=include_inner_1,
-        include_single=include_single,
-        flip_c23=flip_c23,
-        include_triple=include_triple,
-        include_z1=include_z1,
-    )
-
-    guide_trace = prune_subsample_sites(guide_trace)
-    model_trace = prune_subsample_sites(model_trace)
-    model_trace.compute_log_prob()
-    guide_trace.compute_log_prob()
-    non_reparam_nodes = set(guide_trace.nonreparam_stochastic_nodes)
-
-    dc, dc_nodes = _compute_downstream_costs(
```

Comment:
It is better to reimplement `_compute_downstream_costs` in the test using the new approach and keep those tests.

Reply:
Added checking of `_provenance_compute_downstream_costs` to existing tests. In most of the tests `dc_nodes_provenance` is simply `dc_nodes_provenance[name] == {name}` since there is no dependency between variables in these tests except for some observed sites.